src/Microsoft.IdentityModel.Tokens/CryptoProviderFactory.cs

```diff
@@ -102,17 +103,18 @@ public CryptoProviderFactory(CryptoProviderCache cache)
         /// Constructor that creates a deep copy of given <see cref="CryptoProviderFactory"/> object.
         /// </summary>
         /// <param name="other"><see cref="CryptoProviderFactory"/> to copy from.</param>
-        public CryptoProviderFactory(CryptoProviderFactory other)
+        public CryptoProviderFactory(CryptoProviderFactory other) : this()
         {
             if (other == null)
                 throw LogHelper.LogArgumentNullException(nameof(other));
 
-            CryptoProviderCache = new InMemoryCryptoProviderCache() { CryptoProviderFactory = this };
```

Comment:
Where will the cache come from?

Reply:
public CryptoProviderFactory(CryptoProviderFactory other) : this()

Calling the other (default) constructor will call the CryptoProviderCacheFactory to create a default (LRU) cache.
I am trying to remove dependencies and inject them instead.